pagopa/io-app-design-system
# [IACROSS-221] Update the new `FooterWithButtons` to be more reliable

## Short description
This PR improves the behaviour of the `FooterWithButtons` component. In fact, there are two issues in the current iteration:
- The block containing buttons has a safe bottom padding, but in the smartphones with edge-to-edge screens and home indicator (like iPhone X and up), the space at the bottom is too big.
- There's no way to get the footer height value to avoid hiding content (under the component), or placing huge blocks of space to safely display the content
 
## List of changes proposed in this pull request
- Add an `onLayout` callback function to get the correct height value. This will ensure that the `FooterWithButtons` is rendered correctly in both versions (_legacy_ and _new_).
- Add a new screen in the _example app_ to show both cases: long content and short content (with centered content, very common in empty/success/failure screens)

### Preview
| Before | After |
|--------|--------|
| <video src="https://github.com/pagopa/io-app-design-system/assets/1255491/c5d74e45-4728-49e4-bdb4-81cf67c6fcff" /> | <video src="https://github.com/pagopa/io-app-design-system/assets/1255491/38451afd-efc7-49e1-a6dc-2e551aecf604" /> | 




## How to test
1. Run the example app
2. Go to the **Screens** section and check both screens
